### ISSUE TITLE
Ability to cd directly to one-off directories

### ISSUE THREAD

**Issue by fryit5150** [jarun/nnn#1758]
Can you add a way to go directly to a one-off directory? Say if I need to a directory in /usr/share it's quite clumsy to have to type `, ^N, usr, share, ... rather than e.g. :cd <ctrl+shift+v> in other file managers.

**Comment by N-R-K** [comment]
It should be possible to do this quite easily with a plugin. There's already a couple `cd` related plugin you can take a look at for reference.

**Comment by fryit5150** [comment]
It's quite basic functionality, so it would be great not needing to install and setup a plugin for it on all my computers/new installs in VMs.

**Comment by 0xACE** [comment]
> It's quite basic functionality, so it would be great not needing to install and setup a plugin for it on all my computers/new installs in VMs.

It serves no purpose, you can literally hit these keys:

`!mkdir -p foo/bar/baz<ENTER><CTRL-D>`

for a better and more robust functionality...

**Comment by fryit5150** [comment]
> > It's quite basic functionality, so it would be great not needing to install and setup a plugin for it on all my computers/new installs in VMs.
> 
> It serves no purpose, you can literally hit these keys:
> 
> `!mkdir -p foo/bar/baz<ENTER><CTRL-D>`
> 
> for a better and more robust functionality...

You are right. Running mkdir to browse a directory makes no sense at all.

**Comment by N-R-K** [comment]
Here's a plugin to do what you want:

```sh
#!/bin/sh

printf "enter path to cd: "
read -r to
[ -n "$to" ] && printf "0c${to}" > "$NNN_PIPE"
```

> it would be great not needing to install and setup a plugin for it on all my computers/new installs in VMs

Whether to include this feature in nnn natively or not is @jarun's call. I don't really have any preference here one way or other.

**Comment by 0xACE** [comment]
Sorry i didn't fully read the topic.

This is the way it was implemented on my branch: when in filter mode: `/` will cd into a directory if the highlighted line is a directory otherwise it will clear the filter + show all files and highlight the matched item, essentially allowing you to paste a path to go to the desired location.

I just gave it a try, it still works after 3 years, but I guess it doesnt work while `nnn` is running within `tmux`...

**Comment by jarun** [comment]
We can't add a new keybind for changing directories.
Plugins are one of the best features of `nnn` and very simple to migrate and use.

However, in this case, did you try to add the following in your ~/.bashrc:

    NNN_PLUG='c:!read -r to && [ -n "$to" ] && printf "0c%s" "${to}" > "$NNN_PIPE"*'

and press <kbd>Alt-C</kbd>?

**Comment by jarun** [comment]
Fixed at 9259170afd82943180b58cd0e645a54cb3a614be.
